src/main/java/org/elasticsearch/gateway/MetaDataStateFormat.java

```diff
@@ -83,15 +85,15 @@ public XContentType format() {
      * it's target filename of the pattern <tt>{prefix}{version}.st</tt>.
      *
      * @param state the state object to write
-     * @param prefix the state names prefix used to compose the file name.
      * @param version the version of the state
      * @param locations the locations where the state should be written to.
      * @throws IOException if an IOException occurs
      */
-    public final void write(final T state, final String prefix, final long version, final Path... locations) throws IOException {
+    public final void write(final T state, final long version, final Path... locations) throws IOException {
         Preconditions.checkArgument(locations != null, "Locations must not be null");
         Preconditions.checkArgument(locations.length > 0, "One or more locations required");
-        String fileName = prefix + version + STATE_FILE_EXTENSION;
+        final long maxStateId = Math.max(findMaxStateId(prefix, locations)+1, version);
```

Comment:
I really want to prevent yet another file name format. I think one ID is fine we can also just use `-1` as the base and increment it on each new state. but not another format please. I think it really doesn't matter since the ID is only used to find the latest nothing todo with the actual version